pkg/cmd/release-payload-controller/aggregate_state_controller.go

```diff
@@ -0,0 +1,113 @@
+	releasepayloadinformer "github.com/openshift/release-controller/pkg/client/informers/externalversions/release/v1alpha1"
+	"github.com/openshift/release-controller/pkg/releasepayload/jobrunresult"
+	"github.com/openshift/release-controller/pkg/releasepayload/jobstatus"
+	releasepayloadhelpers "github.com/openshift/release-controller/pkg/releasepayload/v1alpha1helpers"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	utilruntime "k8s.io/apimachinery/pkg/util/runtime"
+	"k8s.io/client-go/tools/cache"
+	"k8s.io/client-go/util/workqueue"
+	"k8s.io/klog/v2"
+	"reflect"
+
+	"github.com/openshift/library-go/pkg/operator/events"
+)
+
+// AggregateStateController is responsible for updating the AggregateState for every entry in the
+// BlockingJobResults and InformingJobResults lists.
+// The AggregateStateController reads the following pieces of information:
+//   - .status.blockingJobResults[] | .results[]
+//   - .status.informingJobResults[] | .results[]
+// and populates the following condition:
+//   - .status.blockingJobResults[] | .state
+//   - .status.informingJobResults[] | .state
+type AggregateStateController struct {
+	*ReleasePayloadController
+}
+
+func NewAggregateStateController(
+	releasePayloadNamespace string,
+	releasePayloadInformer releasepayloadinformer.ReleasePayloadInformer,
+	releasePayloadClient releasepayloadclient.ReleaseV1alpha1Interface,
+	eventRecorder events.Recorder,
+) (*AggregateStateController, error) {
+	c := &AggregateStateController{
+		ReleasePayloadController: NewReleasePayloadController("Aggregate State Controller",
+			releasePayloadNamespace,
+			releasePayloadInformer,
+			releasePayloadClient,
+			eventRecorder.WithComponentSuffix("aggregate-state-controller"),
+			workqueue.NewNamedRateLimitingQueue(workqueue.DefaultControllerRateLimiter(), "AggregateStateController")),
+	}
+
+	c.syncFn = c.sync
+
+	releasePayloadInformer.Informer().AddEventHandler(&cache.ResourceEventHandlerFuncs{
+		AddFunc: c.Enqueue,
+		UpdateFunc: func(oldObj, newObj interface{}) {
+			c.Enqueue(newObj)
+		},
+		DeleteFunc: c.Enqueue,
+	})
+
+	return c, nil
+}
+
+func (c *AggregateStateController) sync(ctx context.Context, key string) error {
+	klog.V(4).Infof("Starting AggregateStateController sync")
+	defer klog.V(4).Infof("AggregateStateController sync done")
+
+	// Convert the namespace/name string into a distinct namespace and name
+	namespace, name, err := cache.SplitMetaNamespaceKey(key)
+	if err != nil {
+		utilruntime.HandleError(fmt.Errorf("invalid resource key: %s", key))
+		return nil
+	}
+
+	// Get the ReleasePayload resource with this namespace/name
+	originalReleasePayload, err := c.releasePayloadLister.ReleasePayloads(namespace).Get(name)
+	// The ReleasePayload resource may no longer exist, in which case we stop processing.
+	if errors.IsNotFound(err) {
+		return nil
+	}
+	if err != nil {
+		return err
+	}
+
+	releasePayload := originalReleasePayload.DeepCopy()
```

Comment:
at larger scales, you will need to figure out how to know if you need to change a value *before* the deepcopy.

Reply:
I've added a `TODO` here and I'll revisit in a follow on PR.